### PR TITLE
fix(extmarks): fix visual highlight continuing after "combine" virtual text

### PR DESCRIPTION
**Pull request by SleepySwords**
This only happens if a "replace" virtual text is visually selected and is before the "combine" virtual text. Should other saved attributes (such as `saved_decor_attr` and `saved_search_attr`) also be reset after they are restored? Just to be safe?

### CODE REVIEW

**Comment by zeertzjq** [comment]
`search_attr` and `decor_attr` are set again in the next loop iteration if it's a buffer character, so the same issue doesn't apply to them, but it is definitely less strange if saved values are cleared when they are restored.

**Comment by SleepySwords** [comment]
Alright, in that case, I'll add clearing `saved_ search_attr` and `saved_ decor_attr`

File: test/functional/ui/decorations_spec.lua [hunk 1]
```diff
@@ -1596,6 +1596,7 @@ describe('decorations: inline virtual text', function()
       [18] = {background = Screen.colors.LightGrey, foreground = Screen.colors.Red};
       [19] = {background = Screen.colors.Yellow, foreground = Screen.colors.SlateBlue};
       [20] = {background = Screen.colors.LightGrey, foreground = Screen.colors.SlateBlue};
+      [21] = {reverse = true, foreground = Screen.colors.SlateBlue}
```
Comment (zeertzjq) [inline test/functional/ui/decorations_spec.lua:1599]: This looks strange. What does this "reverse" come from?

Comment (SleepySwords) [reply test/functional/ui/decorations_spec.lua:1599]: The first combine virtual text is in the "main" search selection which does "reverse" the highlighting of the text.

**Pull request merged**